### DISCUSSION TITLE
Export X,Y,Z coordinates after G38.2

### DISCUSSION THREAD

**Discussion by LarryDR** [winder/Universal-G-Code-Sender#2645] [Help]
I have a project that I need to Probe a non rectangular - trapezoidal or triangular shaped area. I have been using the autoleveler and it works well for a rectangular grid of points, but I would like to be able to either run the AutoLeveler with user specified x,y,z coordinates or I could write a macro to go probe the locations I want but then how do I automatically output the stored 3 point coordinates and then be able to load that text file back into the autoleveler and apply gcode to it? I know I could copy and paste them from the console, but I would like to be able to collect and output them similar to what the Autoleveler does.

**Comment by breiler** [comment]
What do you think about having a option to only scan within the outline of the loaded gcode?

**Reply by LarryDR** [reply]
yea, I think that would work, it would allow the user to make any type of customized probing routine. It would solve my problem as well as open up options to reverse engineer objects.
